common/containers/Tabs/ENS/components/NameResolve/components/CountDown.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import moment from 'moment';
+
+class CountDown extends Component<
+  {
+    initialTime: number;
+  },
+  {
+    currentTime: number;
+  }
+> {
+  public state = { currentTime: 0 };
+  constructor() {
+    super();
+    this.startCountDown();
+  }
+
+  public render() {
+    const { currentTime } = this.state;
+    return <p>{this.humanizeTime(currentTime)}</p>;
+  }
+
+  private humanizeTime = (time: number) => {
+    let timeRemaining = time;
+    const floorTime = unit => Math.floor(timeRemaining / unit);
+    const pad = (num: number) => num.toString().padStart(2, '0');
+    const second = 1000;
+    const minute = second * 60;
+    const hour = minute * 60;
+    const day = hour * 24;
+
+    const days = floorTime(day);
+    timeRemaining -= days * day;
+    const hours = floorTime(hour);
+    timeRemaining -= hours * hour;
+    const minutes = floorTime(minute);
+    timeRemaining -= minutes * minute;
+    const seconds = floorTime(second);
+
+    return `${pad(days)} Days ${pad(hours)} Hours ${pad(minutes)} Minutes ${pad(
+      seconds
+    )} Seconds `;
+  };
```

Comment:
If we end up with much more time displays, I'd much prefer a library for this kind of stuff (Like Moment) rather than custom implementations.

Reply:
Oh wait, we already have Moment. Can we just use that here?

Reply:
I couldn't get the custom formatting I wanted with moment, looked like a needed a countdown plugin for it too.